Tests/Schema.NET.Test/ThingTest.cs

```diff
@@ -199,6 +199,135 @@ public void ToStringWithNullAssignedProperty_ReturnsExpectedJsonLd()
             Assert.Equal(expectedJson, thing.ToString());
         }
 
+        [Fact]
+        public void TrySetValue_ValidProperty()
+        {
+            var thing = new Thing();
+
+            Assert.True(thing.TrySetValue("Name", new[] { "TestName" }));
+            Assert.Equal("TestName", thing.Name);
+        }
+
+        [Fact]
+        public void TrySetValue_InvalidProperty()
+        {
+            var thing = new Thing();
+
+            Assert.False(thing.TrySetValue("InvalidName", new[] { "TestName" }));
+        }
+
+        [Fact]
+        public void TrySetValue_CaseInsensitive()
+        {
+            var thing = new Thing();
+
+            Assert.True(thing.TrySetValue("name", new[] { "TestName" }));
```

Comment:
What made you go with case insensitivity?

Reply:
With the idea you don't know the properties at compile time, it seemed like the best idea for compatibility to support it.

Happy to have it be a setting you pass in instead - can default to case sensitive.